[Backported to Helium M145]

From 5c4c1dbd6bf1e67796a2f6460f593f5bc72a4d80 Mon Sep 17 00:00:00 2001
From: David Pennington <dpenning@chromium.org>
Date: Wed, 04 Feb 2026 17:58:38 -0800
Subject: [PATCH] [Vertical Tabs] Fix context menu below NTB

There was no hit test for the bottom container, however, the bottom container now spans a majority of the VTS so adding in a hit tests fixes the caption area check.

Fixed: 481461371, 481413802
Change-Id: Icef86147d2ebd86a19c75fb23c4774dc42263bbe
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/7543081
Reviewed-by: Kunal Daftari <kunaldaftari@google.com>
Commit-Queue: David Pennington <dpenning@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1579815}
---

--- a/chrome/browser/ui/views/frame/vertical_tab_strip_region_view.cc
+++ b/chrome/browser/ui/views/frame/vertical_tab_strip_region_view.cc
@@ -383,6 +383,10 @@ bool VerticalTabStripRegionView::IsPosit
       if (child == tab_strip_view_) {
         return tab_strip_view_->IsPositionInWindowCaption(point_in_child);
       }
+      if (child == bottom_button_container_) {
+        return bottom_button_container_->IsPositionInWindowCaption(
+            point_in_child);
+      }
       return false;
     }
   }
--- a/chrome/browser/ui/views/tabs/vertical/vertical_tab_strip_bottom_container.cc
+++ b/chrome/browser/ui/views/tabs/vertical/vertical_tab_strip_bottom_container.cc
@@ -103,6 +103,21 @@ void VerticalTabStripBottomContainer::Sh
   everything_menu_->RunMenu();
 }
 
+bool VerticalTabStripBottomContainer::IsPositionInWindowCaption(
+    const gfx::Point& point) {
+  for (views::View* child : children()) {
+    if (!child->GetVisible()) {
+      continue;
+    }
+    gfx::Point point_in_child = point;
+    views::View::ConvertPointToTarget(this, child, &point_in_child);
+    if (child->HitTestPoint(point_in_child)) {
+      return false;
+    }
+  }
+  return true;
+}
+
 void VerticalTabStripBottomContainer::OnCollapsedStateChanged(
     tabs::VerticalTabStripStateController* controller) {
   UpdateButtonStyles(controller);
--- a/chrome/browser/ui/views/tabs/vertical/vertical_tab_strip_bottom_container.h
+++ b/chrome/browser/ui/views/tabs/vertical/vertical_tab_strip_bottom_container.h
@@ -38,6 +38,8 @@ class VerticalTabStripBottomContainer :
 
   void ShowEverythingMenu();
 
+  bool IsPositionInWindowCaption(const gfx::Point& point);
+
   void OnCollapsedStateChanged(
       tabs::VerticalTabStripStateController* state_controller);
 
